### PR TITLE
Feat: twitter skills

### PR DESCRIPTION
**Pull request by hyacinthus**
# Pull Request Template

## Description
Twitter skills

## Type of Change
- [ ] Bugfix
- [x] New Feature
- [ ] Improvement
- [ ] Documentation Update

## Checklist
- [x] I have read the contributing guidelines.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## Related Issue
Closes #[issue number]

### CODE REVIEW

File: app/core/ai.py [hunk 1]
```diff
@@ -58,101 +61,102 @@ def initialize_agent(aid):
...
+
+        prompt = ""
+        if agent.name:
+            prompt = f"Your name is {agent.name}. "
+
+        if agent.prompt:
+            prompt += agent.prompt
+        elif agent.cdp_enabled:
+            prompt += (
+                "You are a helpful agent that can interact onchain using the Coinbase Developer Platform AgentKit. "
```
Comment (taiyangc) [inline app/core/ai.py:142]: we should remove this a bit later to make it more extensible
essentially, each skill should have some default "system prompt" that safeguards things

File: app/models/agent.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    # if cdp_enabled, will load cdp skills
+    # if the cdp_skills is empty, will load all
+    cdp_enabled: bool = Field(default=False)
+    cdp_skills: Optional[List[str]] = Field(sa_column=Column(JSONB, nullable=True))
+    cdp_wallet_data: Optional[str]
+    cdp_network_id: Optional[str]
+    # if twitter_enabled, twitter_config will be checked
+    twitter_enabled: bool = Field(default=False)
+    twitter_config: Optional[dict] = Field(sa_column=Column(JSONB, nullable=True))
+    twitter_skills: Optional[List[str]] = Field(sa_column=Column(ARRAY(String)))
```
Comment (taiyangc) [inline app/models/agent.py:34]: the skill "categories" should be more generic, for example, should be social_skills -> twitter_skills

**Review by taiyangc — APPROVED** [review]
nothing blocking merge, but recommendations for later improvement

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -3,84 +3,149 @@ name: Build Docker Image
...
-        env:
-          SLACK_BOT_TOKEN: ${{ secrets.SLACK_BOT_TOKEN }}
+    - name: Build Start
+      id: ci_start
+      uses: slackapi/slack-github-action@485a9d42d3a73031f12ec201c457e2162c45d02d
+      with:
+        method: chat.postMessage
+        token: ${{ secrets.SLACK_BOT_TOKEN }}
+        payload: |
+          channel: 'C0786MHAL8J'
```
Comment (taiyangc) [inline .github/workflows/build.yml:28]: this should be a config var

Comment (taiyangc) [reply .github/workflows/build.yml:28]: maybe shouldnt be in `main`

Comment (hyacinthus) [reply .github/workflows/build.yml:28]: I'll add the configuration at the repository level. What does "maybe shouldn't be in main" mean?

File: app/core/ai.py (hunk 1 above)
Comment (hyacinthus) [reply app/core/ai.py:142]: Each skill do have their prompt, they will pass to model too. This one only takes effect when the user has not provided a custom prompt and the CDP tool is enabled. We can come up with something else to replace it in future updates.

File: app/models/agent.py (hunk 2 above)
Comment (hyacinthus) [reply app/models/agent.py:34]: I haven't figured out how to organize the configuration for general skills and skill sets yet. In any case, I want to treat Twitter and Telegram as "first-class" configurations; they are special and at the root level of the configuration. For example, Slack is configured within the skill sets and is not visible in the configuration definitions.

File: .github/workflows/build.yml (hunk 3 above)
Comment (taiyangc) [reply .github/workflows/build.yml:28]: this is a centralized slack notification decision - it's not part of intentkit core build

Comment (taiyangc) [reply .github/workflows/build.yml:28]: but saving at repo level is ok for now

**Review by taiyangc — APPROVED** [review]

**Pull request merged**